src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ActionAttributeRouteModel.cs

```diff
@@ -3,11 +3,141 @@
 
 using System.Collections.Generic;
 using System.Linq;
+using Microsoft.AspNetCore.Mvc.ActionConstraints;
+using Microsoft.AspNetCore.Mvc.Routing;
```

Comment:
As a personal note.. Most of this code was written by me from like 4 years ago. It was a little bit fun to revisit with some more wisdom (I hope) - I feel like I was about to improve upon some code I wrote with old habits.